crates/proto/src/xfer/dnssec_dns_handle/nsec3_validation.rs

```diff
@@ -70,11 +70,22 @@
 use crate::{
     op::{Query, ResponseCode},
     rr::{
-        dnssec::{rdata::NSEC3, Proof},
+        dnssec::{rdata::NSEC3, Nsec3HashAlgorithm, Proof},
         Name, Record, RecordType,
     },
 };
 
+fn nsec3hash(name: &Name, salt: &[u8], iterations: u16) -> Vec<u8> {
+    Nsec3HashAlgorithm::SHA1
```

Comment:
I doubt we will ever see NSEC3 using any hashing algorithm that's not SHA-1 but to be consistent I think you should take `Nsec3HashAlgorithm` here as a parameter as well.

Reply:
I made this function specifically to keep call sites clean. I see no point making them too complicated with no gain.